src/Compilers/CSharp/Test/Syntax/Parsing/FieldKeywordParsingTests.cs

```diff
@@ -0,0 +1,2006 @@
+        {
+            IdentifierNameOrFieldExpression(languageVersion, escapeIdentifier, IsParsedAsToken(languageVersion, escapeIdentifier));
+        }
+
+        private void IdentifierNameOrFieldExpression(LanguageVersion languageVersion, bool escapeIdentifier, bool isParsedAsToken)
+        {
+            if (isParsedAsToken)
+            {
+                N(SyntaxKind.FieldExpression);
+                {
+                    N(SyntaxKind.FieldKeyword);
+                }
+            }
+            else
+            {
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, GetFieldIdentifier(escapeIdentifier));
+                }
+            }
+        }
+
+        private static string GetFieldIdentifier(bool escapeIdentifier)
+        {
+            return escapeIdentifier ? "@field" : "field";
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void Property_Initializer(
+            [CombinatorialValues(LanguageVersion.CSharp12, LanguageVersion.Preview)] LanguageVersion languageVersion)
+        {
+            bool expectedParsedAsToken = false;
+            UsingTree($$"""
+                class C
+                {
+                    object P { get; set; } = field;
+                }
+                """,
+                TestOptions.Regular.WithLanguageVersion(languageVersion));
+
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxKind.ClassDeclaration);
+                {
+                    N(SyntaxKind.ClassKeyword);
+                    N(SyntaxKind.IdentifierToken, "C");
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.PropertyDeclaration);
+                    {
+                        N(SyntaxKind.PredefinedType);
+                        {
+                            N(SyntaxKind.ObjectKeyword);
+                        }
+                        N(SyntaxKind.IdentifierToken, "P");
+                        N(SyntaxKind.AccessorList);
+                        {
+                            N(SyntaxKind.OpenBraceToken);
+                            N(SyntaxKind.GetAccessorDeclaration);
+                            {
+                                N(SyntaxKind.GetKeyword);
+                                N(SyntaxKind.SemicolonToken);
+                            }
+                            N(SyntaxKind.SetAccessorDeclaration);
+                            {
+                                N(SyntaxKind.SetKeyword);
+                                N(SyntaxKind.SemicolonToken);
+                            }
+                            N(SyntaxKind.CloseBraceToken);
+                        }
+                        N(SyntaxKind.EqualsValueClause);
+                        {
+                            N(SyntaxKind.EqualsToken);
+                            IdentifierNameOrFieldExpression(languageVersion, escapeIdentifier: false, expectedParsedAsToken);
```

Comment:
It feels like it would be more straightforward to just use the generated baseline when we expect it to be the same in all the cases of the theory.